graylog2-web-interface/src/components/alertnotifications/AlertNotification.jsx

```diff
@@ -35,13 +37,21 @@ const AlertNotification = React.createClass({
 
   _onSubmit(data) {
     AlarmCallbacksActions.update(this.props.alertNotification.stream_id, this.props.alertNotification.id, data)
-      .then(() => AlertNotificationsActions.listAll());
+      .then(() => {
+        if (typeof this.props.onNotificationUpdate === 'function') {
```

Comment:
Wouldn't a simple check for undefined/null be enough here since we specify the propTypes?